api/src/flows.ts

```diff
@@ -0,0 +1,358 @@
+					if (flow.options.async) {
+						this.executeFlow(flow, data, context);
+					} else {
+						return this.executeFlow(flow, data, context);
+					}
+				};
+
+				const method = flow.options?.method ?? 'GET';
+
+				// Default return to $last for webhooks
+				flow.options.return = flow.options.return ?? '$last';
+
+				this.webhookFlowHandlers[`${method}-${flow.id}`] = handler;
+			}
+		}
+	}
+
+	public async reload(): Promise<void> {
+		for (const trigger of this.triggerHandlers) {
+			trigger.events.forEach((event) => {
+				switch (event.type) {
+					case 'filter':
+						emitter.offFilter(event.name, event.handler);
+						break;
+					case 'action':
+						emitter.offAction(event.name, event.handler);
+						break;
+					case 'schedule':
+						event.task.stop();
+						break;
+				}
+			});
+		}
+
+		this.triggerHandlers = [];
+		this.operationFlowHandlers = {};
+		this.webhookFlowHandlers = {};
+
+		this.flowOperations = {};
+
+		await this.initialize();
+	}
+
+	public addOperation(id: string, operation: OperationHandler): void {
+		this.operations[id] = operation;
+	}
+
+	public clearOperations(): void {
+		this.operations = {};
+	}
+
+	public async runOperationFlow(id: string, data: unknown, context: Record<string, unknown>): Promise<unknown> {
+		if (!(id in this.operationFlowHandlers)) {
+			logger.warn(`Couldn't find operation triggered flow with id "${id}"`);
+			return null;
+		}
+
+		const handler = this.operationFlowHandlers[id];
+
+		return handler(data, context);
+	}
+
+	public async runWebhookFlow(id: string, data: unknown, context: Record<string, unknown>): Promise<unknown> {
+		if (!(id in this.webhookFlowHandlers)) {
+			logger.warn(`Couldn't find webhook triggered flow with id "${id}"`);
+			throw new exceptions.ForbiddenException();
+		}
+
+		const handler = this.webhookFlowHandlers[id];
+
+		return handler(data, context);
+	}
+
+	public async runOperation(id: string, data: unknown): Promise<unknown> {
+		if (!(id in this.flowOperations)) {
+			logger.warn(`Couldn't find operation with id "${id}"`);
+
+			return null;
+		}
+
+		const handler = this.flowOperations[id];
+
+		return handler(data);
+	}
+
+	private async executeFlow(flow: Flow, data: unknown = null, context: Record<string, unknown> = {}): Promise<unknown> {
+		const database = (context.database as Knex) ?? getDatabase();
+		const schema = (context.schema as SchemaOverview) ?? (await getSchema({ database }));
+
+		const keyedData: Record<string, unknown> = {
+			[TRIGGER_KEY]: data,
+			[LAST_KEY]: data,
+			[ACCOUNTABILITY_KEY]: context?.accountability ?? null,
+		};
+
+		let nextOperation = flow.operation;
+
+		while (nextOperation !== null) {
+			const { successor, data } = await this.executeOperation(nextOperation, keyedData, context);
+
+			keyedData[nextOperation.key] = data;
+			keyedData[LAST_KEY] = data;
+
+			nextOperation = successor;
+		}
+
+		if (flow.accountability !== null) {
```

Comment:
The name of this option might be a bit confusing. To me `accountability` indicates that this option would somehow alter the accountability object of each operation associated with the flow, ehile it actually only changes the activity tracking. Maybe we could rename this to something like `tracking`?

Reply:
While I agree, we're using the same nomenclature on `directus_collections` currently, so I'd rather keep it consistent. We're can still surface it as "tracking" in the UI though (just like for collections):

<img width="468" alt="CleanShot 2022-05-27 at 12 47 02@2x" src="https://user-images.githubusercontent.com/9141017/170743346-9e5e112f-53d1-49ee-98c2-67118f86b9d1.png">